internal/notifier/opsgenie.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"crypto/x509"
+	"fmt"
+	"net/url"
+
+	"github.com/fluxcd/pkg/runtime/events"
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type Opsgenie struct {
+	URL      string
+	ProxyURL string
+	CertPool *x509.CertPool
+	ApiKey   string
+}
+
+type OpsgenieAlert struct {
+	Message     string            `json:"message"`
+	Description string            `json:"description"`
+	Details     map[string]string `json:"details"`
+}
+
+// NewSlack validates the Slack URL and returns a Slack object
+func NewOpsgenie(hookURL string, proxyURL string, certPool *x509.CertPool, token string) (*Opsgenie, error) {
+	_, err := url.ParseRequestURI(hookURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid Opsgenie hook URL %s", hookURL)
+	}
+
+	return &Opsgenie{
+		URL:      hookURL,
+		ProxyURL: proxyURL,
+		CertPool: certPool,
+		ApiKey:   token,
```

Comment:
Is the token required? If so we should error out if it's an empty string. If this is required, I think we should add an Opsgenie section to the API docs and show people how to create the secret and which keys to set.

Reply:
Thank you!